assets/js/worker.js

```diff
@@ -1,18 +1,38 @@
-onmessage = function(event) {
-	self.importScripts('https://unpkg.com/lunr/lunr.js');
-	
+let scriptLoaded = false;
+
+onmessage = async function(event) {
+	const scriptUrl = 'https://unpkg.com/lunr/lunr.js';
+  const integrityHash = 'sha256-lDFybwXA6uKm5U3Bl3CUIoafJcrUTyQw0vt92ugMxxc=';
```

Comment:
⚠️⚠️⚠️ Warning, this is brittle and should be upgraded! ⚠️⚠️⚠️ 

When you hardcode an integrity hash value, you need to make sure the asset is static and deterministic. The url in this case contains no version information, which means it loads the latest published version and can change, which would cause the integrity check to fail.

Even without the integrity check btw, best practice is to use static-version assets, so you never have a case where an expected upgrade breaks something.

In this case url should be:
https://unpkg.com/lunr@2.3.9/lunr.js

btw, while you are at it, might as well use the minified version:
https://unpkg.com/lunr@2.3.9/lunr.min.js


Minor side nit to that: it could be good to put the computation of the integrity hashes in a script somewhere, so people know what to do when they upgrade the urls in the future. Computing the hashes and injecting them could even be a build step.